force-app/main/domain/Addresses.cls

```diff
@@ -292,6 +292,10 @@ public inherited sharing class Addresses extends fflib_SObjects2 {
                 NPSP_Contact npspContact = new NPSP_Contact(contact);
 
                 Boolean shouldUpdateContact= false;
+                if (npspContact.hasAddressOverride()) {
```

Comment:
There's logic below that possibly contradicts this.

On line 301-3, it will set IsAddressOverride to false if the Contact.CurrentAdress is null. So, if this IsAddressOverride is checked and there is no lookup to a CurrentAddress for the Contact, is the new intention that we'll leave the IsAddressOverride field as checked?

If that's true, I suspect some of the code below can be adjusted to support that new behavior. However, I'm also uncertain if that's what the exact intent is. 

Is there valid reason to lave Is_Address_Override__c set to true if there is no current address on the Contact? There could be; in that maybe the person just has no address for some reason. If true, then let's clean up the contradicting code below.

Reply:
Thanks for the suggestion, @force2b. You're right. Since we already use
'continue' when the contact has an address override, the logic to set 'Is_Address_Override__c' to false when 'CurrentAddress' is null is redundant. I'll remove that part to avoid unnecessary processing and ensure the behavior aligns with the intention of skipping updates for override-enabled contacts.